### DISCUSSION TITLE
Are you planning to switch to the route strategy for a simple configuration file? or to have the possibility of using one or the other?

### DISCUSSION THREAD

**Discussion by milabron** [vercel/next.js#19378] [Help]
I have realized that the `directory `routing system is only for very basic things, and I see more and more new things by trying to simulate a thing that would be fixed by just having a `route file` and saying which component/page to use and with which path. The complexity and mess that the Directory is adding has no limits.

Please let me know if you plan to add this to look for other solutions or if it is possible to try to help and if a pull request for a file routing system will be accepted?

routes.js

```
const routes = [
     {
         name: 'index',
          path: "/",
          page/component: "/",
          params: [],
          redirect: ''
        },
```

with a simple route file you save everything:

1. exportPathMap
2. getStaticPaths
3. rewrites
4. redirects
5. directory complexity

etc...

**Comment by milabron** [comment]
Something like this does not exist on Next.js? https://nuxtjs.org/docs/2.x/features/file-system-routing#extending-the-router

**Comment by rafaelalmeidatk** [comment]
You are free to create a RFC but it would be better if you could give an example of what you are trying accomplish

**Comment by milabron** [comment]
Hello, Thank you for responding.

The only thing I try to do is to reuse my pages/components, I have tried to use rewrites, but the same thing happens to me as this issue https://github.com/vercel/next.js/issues/19382
